### PR TITLE
Clean task no longer fails silently

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Improve `_destroy_stack` function to not fails silently and explicitly suggest adding `aws-vault exec` when the credentials are not found

### CODE REVIEW

**Review by pducolin — APPROVED** [review]

**Comment by pducolin** [comment]
We could even add the wrapper,  one feedback we are getting is that it is not straightforward to add the wrapper

**Comment by KevinFairise2** [comment]
> We could even add the wrapper, one feedback we are getting is that it is not straightforward to add the wrapper

Agree, just created https://datadoghq.atlassian.net/jira/software/c/projects/ADXT/boards/6196/backlog?epics=visible&selectedIssue=ADXT-356 to tackle that

File: tasks/new_e2e_tests.py [hunk 1]
```diff
@@ -288,13 +288,21 @@ def _destroy_stack(ctx: Context, stack: str):
             env=_get_default_env(),
         )
         if ret is not None and ret.exited != 0:
+            if "No valid credential sources found" in ret.stdout:
+                print(
+                    "No valid credentials sources found, you need to wrap the invoke command in aws-vault exec <account> --"
+                )
+                exit(1)
```
Comment (amenasria) [inline tasks/new_e2e_tests.py:295]:
```suggestion
                raise Exit(color_message("No valid credentials sources found, you need to wrap the invoke command in aws-vault exec <account> --", Color.RED), code=1)
```

**💬 suggestion:** ‏For a cleaner output, feel free to ignore.

File: tasks/new_e2e_tests.py [hunk 2]
```diff
@@ -288,13 +288,21 @@ def _destroy_stack(ctx: Context, stack: str):
...
-            ctx.run(
+            ret = ctx.run(
                 f"pulumi destroy --stack {stack} -r --yes --remove --skip-preview",
                 warn=True,
                 hide=True,
                 env=_get_default_env(),
             )
+        if ret is not None and ret.exited != 0:
+            print(f"Failed to destroy stack {stack}, error: {ret.stdout}")
+            exit(1)
```
Comment (amenasria) [inline tasks/new_e2e_tests.py:305]: **💬 suggestion:** ‏Same as above

**Review by amenasria — APPROVED** [review]
LGTM !

File: tasks/new_e2e_tests.py (hunk 2 above)
Comment (KevinFairise2) [reply tasks/new_e2e_tests.py]: Applied

File: tasks/new_e2e_tests.py (hunk 1 above)
Comment (KevinFairise2) [reply tasks/new_e2e_tests.py]: applied

**Pull request merged**